great_expectations/data_context/store/datasource_store.py

```diff
@@ -125,17 +125,6 @@ def retrieve_by_name(self, datasource_name: str) -> DatasourceConfig:
         datasource_config: DatasourceConfig = copy.deepcopy(self.get(datasource_key))  # type: ignore[assignment]
         return datasource_config
 
-    def delete_by_name(self, datasource_name: str) -> None:
-        """Deletes a DatasourceConfig persisted in the store by it's given name.
-
-        Args:
-            datasource_name: The name of the Datasource to retrieve.
-        """
-        datasource_key: DataContextVariableKey = self._determine_datasource_key(
-            datasource_name=datasource_name
-        )
-        self.remove_key(datasource_key)
```

Comment:
Didn't really find a need for this anymore and think standardization between stores is important - let me know if you disagree but IIRC, we wanted to remove these helpers